src/frontend/src/flows/addDevice/welcomeView/registerTentativeDevice.ts

```diff
@@ -1,174 +1,168 @@
-      tentativeDeviceInfo[1],
-      result.added_tentatively,
-      Array.from(new Uint8Array(tentativeDeviceInfo[5]))
-    );
-  } else if ("device_registration_mode_off" in result) {
-    await deviceRegistrationDisabledInfo(connection, tentativeDeviceInfo);
-  } else if ("another_device_tentatively_added" in result) {
+  if (result instanceof Error) {
     await displayError({
-      title: "Tentative Device Already Exists",
-      message:
-        'The "add device" process was already started for another device. If you want to add this device instead, log in using an existing device and restart the "add device" process.',
+      title: "Error adding new device",
+      message: "Unable to register new WebAuthn Device.",
+      detail: result.message,
       primaryButton: "Ok",
     });
     // TODO L2-309: do this without reload
-    window.location.reload();
-  } else {
-    throw new Error(
-      "unknown tentative device registration result: " + JSON.stringify(result)
-    );
+    return window.location.reload() as never;
   }
-};
 
-const init = async (userNumber: bigint, connection: Connection) => {
-  const existingAuthenticators = await withLoader(() =>
-    connection.lookupAuthenticators(userNumber)
-  );
-  const cancelButton = document.getElementById(
-    "registerTentativeDeviceCancel"
-  ) as HTMLButtonElement;
-
-  cancelButton.onclick = () => {
-    // TODO L2-309: do this without reload
-    window.location.reload();
-  };
+  // Finally, we submit it to the canister
+  const device: Omit<DeviceData, "origin"> & { credential_id: [CredentialId] } =
+    {
+      alias: alias,
+      protection: { unprotected: null },
+      pubkey: Array.from(new Uint8Array(result.getPublicKey().toDer())),
+      key_type: { unknown: null },
+      purpose: { authentication: null },
+      credential_id: [Array.from(new Uint8Array(result.rawId))],
+    };
+  const addResponse = await addTentativeDevice({
+    userNumber,
+    connection,
+    device,
+  });
 
-  const continueButton = document.getElementById(
-    "registerTentativeDeviceContinue"
-  ) as HTMLButtonElement;
-  const aliasInput = document.getElementById(
-    "tentativeDeviceAlias"
-  ) as HTMLInputElement;
+  // If everything went well we can now ask the user to authenticate on an existing device
+  // and enter a verification code
+  await showVerificationCode(
+    userNumber,
+    connection,
+    device.alias,
+    addResponse.added_tentatively,
+    device.credential_id[0]
+  );
+};
 
-  aliasInput.onkeypress = (e) => {
-    // submit if user hits enter
-    if (e.key === "Enter") {
-      e.preventDefault();
-      continueButton.click();
+/** Create new WebAuthn credentials */
+const createDevice = async ({
+  userNumber,
+  connection,
+}: {
+  userNumber: bigint;
+  connection: Connection;
+}): Promise<WebAuthnIdentity | Error> => {
+  const existingAuthenticators = await connection.lookupAuthenticators(
+    userNumber
+  );
+  let newDevice: WebAuthnIdentity;
```

Comment:
not sure you need this variable

```
try {
   return await WebAuthnIdentity.create({
      publicKey: creationOptions(existingAuthenticators),
    });
    ...
```

might do

Reply:
Ah good point!

Reply:
Changed 👍